### ISSUE TITLE
ignore_swiftui_view_bodies false positive in conditions

### ISSUE THREAD

**Issue by rd-david-wahlandt** [realm/SwiftLint#6255]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

When configuring `redundant_discardable_let` with `ignore_swiftui_view_bodies: true` the rule still emits a warning when the `let _ =` is inside  condition:

```swift
// This triggers a violation:
@ViewBuilder func makeView() -> some View {
    if true {
        let _ = assertionFailure("This should not trigger")
    }
}
```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint lint [--no-cache] [--fix]
```

### Environment

* SwiftLint version: 0.61.0
* Xcode version: 16.4
* Installation method used: Mint
* Configuration file:

```yml
redundant_discardable_let:
  ignore_swiftui_view_bodies: true
```
